pkg/model/model.go

```diff
@@ -90,10 +93,77 @@ func Load(modelURI, dst string, db *database.DB) (*Model, error) {
 		} else {
 			return nil, fmt.Errorf("error modelURI format: %s", modelURI)
 		}
+	} else if strings.Contains(modelURI, "/") {
+		// general model zoo urls like some-domain.com:port/model_name:tag
+		// download traind model and extract to dst
+		return loadModelFromModelZoo(modelURI, dst)
 	}
 	return loadDB(db, modelURI, dst)
 }
 
+func loadModelFromModelZoo(modelURI, dst string) (*Model, error) {
+	// modelURI is formated like some-domain.com:port/model_name:tag
+	uriParts := strings.Split(modelURI, "/")
+	// previous checks asserted that uriParts must have two parts
+	modelZooServerAddr := uriParts[0]
+	modelNameAndTag := strings.Join(uriParts[1:], "/")
+	parts := strings.Split(modelNameAndTag, ":")
+	var modelName string
+	var modelTag string
+	if len(parts) == 1 {
+		modelName = modelNameAndTag
+		modelTag = "" // default use empty tag
+	} else if len(parts) == 2 {
+		modelName = parts[0]
+		modelTag = parts[1]
+	} else {
+		return nil, fmt.Errorf("model name after USING must be like [model-zoo.com:port/]model_name[:tag]")
+	}
+	// TODO(typhoonzero): use SSL connection
+	conn, err := grpc.Dial(modelZooServerAddr, grpc.WithInsecure())
```

Comment:
Should the model zoo service a gRPC (backend) service, or an HTTP ( frontend) service?

Reply:
model zoo server is a gRPC server to provide core functions, and we can use https://github.com/grpc/grpc-web to provide a front end web pages for users to view available models in the model zoo

Reply:
Should model.go connect to an HTTP(s) proxy of the model zoo gRPC service, other than connecting directly to the gRPC service?  My purpose is that the use of model zoo service should be very flexible.  For example, an on-premise deployment of the SQLFlow service might (optionally) use a model zoo service deployment on public clouds.

Reply:
For on-premise deployments, we can deploy some TCP reverse proxies to proxy internal traffic out to a public model zoo server. Or we can configure the firewall to allow traffic to the specific domain of the public model zoo server. And if we make the model zoo an HTTP service, this type of work is still needed if there's traffic controls in the on-premise environment.